### DISCUSSION TITLE
When the gradle plugin is going to be compatible with kotlin 2.0.0

### DISCUSSION THREAD

**Discussion by vcruzmj** [detekt/detekt#7367] [Q&A] [answered]
When the Gradle plugin is going to be compatible with kotlin 2.0.0, the current approach described in https://detekt.dev/docs/gettingstarted/gradle#gradle-runtime-dependencies is not working for kotlin 2.0.0

**Comment by BraisGabin** [comment, marked as answer]
It is compatible. If you find a bug, open an issue or if you want support al on the slack channel or here but providing enoght information to help you.

**Comment by vcruzmj** [comment]
Thanks for the answer, I have [this](https://github.com/vcruzmj/demo-detekt) repo as an example with the reason I opened the question, which is just a base spring boot project, with the default initializr libraries and detekt (and kotlin 2.0.0). 

The current approach for mismatching versions of your project and the embedded one in detekt (in the context of Gradle runtime dependencies) is described here: 
https://detekt.dev/docs/gettingstarted/gradle/#dependencies. 

This was working in all versions except 2.0.0

```kotlin
configurations.all {
    resolutionStrategy.eachDependency {
        if (requested.group == "org.jetbrains.kotlin") {
            useVersion(io.gitlab.arturbosch.detekt.getSupportedKotlinVersion())
        }
    }
}
```

With kotlin 1.9.4 or minor (when the detekt version of detekt didn't match) that block code fixes all the `compileKotlin` related tasks.
So you can run without any issues `./gradlew assemble` for example, and obviously `./gradlew detekt` or any combination.

But now in 2.0.0, all the `compileKotlin` related tasks are failing with these errors (except ./gradlew detekt which still works, but isolated):

```sh
Execution failed for task ':compileKotlin'.
> Could not resolve all files for configuration ':detachedConfiguration1'.
   > Failed to transform kotlin-reflect-1.9.23.jar to match attributes {artifactType=classpath-entry-snapshot, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for BuildToolsApiClasspathEntrySnapshotTransform: /home/anonym/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.9.23/ab01e1cc8428c2140891318838ad7039f6fa389a/kotlin-reflect-1.9.23.jar.
         > Artifact org.jetbrains.kotlin:kotlin-build-tools-impl must have version aligned with the version of KGP when compilation via the Build Tools API is disabled.
```

So, technically if I add detekt to my project, with kotlin 2.0.0 I can't compile, assemble, build, etc.

The git repo I have [this](https://github.com/vcruzmj/demo-detekt) , has kotlin 2.0.0 in the `main` branch, and 1.9.4 in the `olderexample` branch

But, this still can be my problem, so I don't want to submit an issue yet.

**Reply by BraisGabin** [reply]
Could you open an issue explaining this? It will be easier to track the problem there.

**Reply by vcruzmj** [reply]
I think is part of this https://github.com/detekt/detekt/issues/7304

**Comment by vcruzmj** [comment]
Finally I found the way to configure this, and make it work without problems:

```kotlin
configurations.matching { it.name == "detekt" }.all {
    resolutionStrategy.eachDependency {
        if (requested.group == "org.jetbrains.kotlin") {
            useVersion(io.gitlab.arturbosch.detekt.getSupportedKotlinVersion())
        }
    }
}
```

https://github.com/detekt/detekt/issues/7384#issuecomment-2177456295


Thanks anyway, for this and this project.
